### PR TITLE
[Azure Cognitive Service Unified Vision] 2023-07-01-preview private preview API

### LINKED ISSUE #25696: [Cognitive Services Computer Vision API ] API Review
New API Review meeting has been requested.

**Service Name**: Cognitive Services Computer Vision API 
**Review Created By**: Kuan Lu
**Review Date**: 09/28/2023 04:00 PM PT
**Onboarding Record**: 
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/25660
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: 

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=ba068bd9-044d-ee11-be6e-000d3a3412d3)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by kuanlu95**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [x] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
**The core change is the addition of annotation data in the dataset registration PUT request.**
image
**And definitions of annotation data
line 2670: Annotation
line 2712: AnnotationData
line 2748: Categories
line 2852: Image
All other parts are exactly same as: 2023-04-01-preview/UnifiedVision.json
Also added some examples**

* Previous API Spec Doc:
* Updated paths: /datasets/{name}

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

fix https://github.com/Azure/azure-rest-api-specs/issues/25696

### CODE REVIEW

File: specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-07-01-preview/UnifiedVision.json [hunk 1]
```diff
@@ -0,0 +1,3462 @@
...
+        "id": {
+          "format": "int32",
+          "description": "Image Id.",
+          "maximum": 1000,
+          "minimum": 1,
+          "type": "integer"
+        },
+        "width": {
+          "format": "double",
+          "description": "Image width in pixels.",
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-07-01-preview/UnifiedVision.json:2871]: pixel be int or double?

File: specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-07-01-preview/UnifiedVision.json [hunk 2]
```diff
@@ -0,0 +1,3462 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Computer Vision Client",
+    "description": "The Computer Vision API provides state-of-the-art algorithms to process images and return information. For example, it can be used to extract text using Read OCR, caption an image using descriptive natural language, detect objects, people, and more.",
+    "version": "2023-07-01-preview"
+  },
+  "paths": {
+    "/imageanalysis:segment": {
```
Comment (heaths) [inline specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-07-01-preview/UnifiedVision.json:9]: If you use the POST action pattern, the action has to be a verb. Seems "/imagesanalysis:analyze" makes more sense given your description.

**Review by heaths — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-04-01-preview/UnifiedVision.json
[the remark this answers is not in the document]
Comment (heaths) [reply specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-04-01-preview/UnifiedVision.json:1]: I didn't initially realize this is an up date. For future services, please take a look at our guidelines.

File: specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-07-01-preview/UnifiedVision.json (hunk 2 above)
Comment (heaths) [reply specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-07-01-preview/UnifiedVision.json:9]: I see this is an old endpoint, but my concern still stands but is moot now. Please consider for future services.

File: specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-07-01-preview/UnifiedVision.json [hunk 3]
```diff
@@ -0,0 +1,3462 @@
...
+          "Segment_BackgroundRemoval_FromImageUrl": {
+            "$ref": "./examples/Segment_BackgroundRemoval_FromImageUrl.json"
+          },
+          "Segment_ForegroundMatting_FromImageUrl": {
+            "$ref": "./examples/Segment_ForegroundMatting_FromImageUrl.json"
+          }
+        }
+      }
+    },
+    "/retrieval:vectorizeText": {
```
[the remark this answers is not in the document]
Comment (heaths) [reply specification/cognitiveservices/data-plane/UnifiedVision/preview/2023-07-01-preview/UnifiedVision.json:80]: I see this is an old endpoint, but my concern still stands but is moot now. Please consider for future services.

**Pull request closed without merging**